cmd/subnetcmd/import_from_api.go

```diff
@@ -0,0 +1,282 @@
+	}
+
+	if genesisFilePath == "" {
+		genesisFilePath, err = app.Prompt.CaptureExistingFilepath("Provide the path to the genesis file")
+		if err != nil {
+			return err
+		}
+	}
+
+	var reply *info.GetNodeVersionReply
+
+	if nodeURL == "" {
+		yes, err := app.Prompt.CaptureYesNo("Have nodes already been deployed to this subnet?")
+		if err != nil {
+			return err
+		}
+		if yes {
+			nodeURL, err = app.Prompt.CaptureString(
+				"Please provide an API URL of such a node so we can query its VM version (e.g. 111.22.33.44:5555)")
+			if err != nil {
+				return err
+			}
+			ctx, cancel := context.WithTimeout(context.Background(), constants.RequestTimeout)
+			defer cancel()
+			infoAPI := info.NewClient(nodeURL)
+			options := []rpc.Option{}
+			reply, err = infoAPI.GetNodeVersion(ctx, options...)
+			if err != nil {
+				return fmt.Errorf("failed to query node - is it running and reachable? %w", err)
+			}
+		}
+	}
+
+	var subnetID ids.ID
+	if subnetIDstr == "" {
+		subnetID, err = app.Prompt.CaptureID("What is the ID of the subnet?")
+		if err != nil {
+			return err
+		}
+	} else {
+		subnetID, err = ids.FromString(subnetIDstr)
+		if err != nil {
+			return err
+		}
+	}
+
+	var pubAPI string
+	switch network {
+	case models.Fuji:
+		pubAPI = constants.FujiAPIEndpoint
+	case models.Mainnet:
+		pubAPI = constants.MainnetAPIEndpoint
+	}
+	client := platformvm.NewClient(pubAPI)
+	ctx, cancel := context.WithTimeout(context.Background(), constants.RequestTimeout)
+	defer cancel()
+	options := []rpc.Option{}
+
+	ux.Logger.PrintToUser("Getting information from the %s network...", network.String())
+
+	chains, err := client.GetBlockchains(ctx, options...)
+	if err != nil {
+		return err
+	}
+
+	var (
+		blockchainID, vmID ids.ID
+		subnetName         string
+	)
+
+	for _, ch := range chains {
+		// NOTE: This supports only one chain per subnet
+		if ch.SubnetID == subnetID {
+			blockchainID = ch.ID
+			vmID = ch.VMID
+			subnetName = ch.Name
+			break
+		}
+	}
+
+	if blockchainID == ids.Empty || vmID == ids.Empty {
+		return fmt.Errorf("subnet ID %s not found on this network", subnetIDstr)
+	}
+
+	ux.Logger.PrintToUser("Retrieved information. BlockchainID: %s, Name: %s, VMID: %s",
+		blockchainID.String(),
+		subnetName,
+		vmID.String(),
+	)
+	// TODO: it's probably possible to deploy VMs with the same name on a public network
+	// In this case, an import could clash because the tool supports unique names only
+
+	genBytes, err := os.ReadFile(genesisFilePath)
+	if err != nil {
+		return err
+	}
+
+	if err = app.WriteGenesisFile(subnetName, genBytes); err != nil {
+		return err
+	}
+
+	vmType := getVMFromFlag()
+	if vmType == "" {
+		subnetTypeStr, err := app.Prompt.CaptureList(
+			"What's this VM's type?",
+			[]string{models.SubnetEvm, models.SpacesVM, models.CustomVM},
+		)
+		if err != nil {
+			return err
+		}
+		vmType = models.VMTypeFromString(subnetTypeStr)
+	}
+
+	sc := &models.Sidecar{
+		Name:            subnetName,
```

Comment:
you should set the token name to `TEST`

Reply:
Optionally you could query for the token name.

Reply:
We'll leave it to the pre-set TEST token for now. We can address this later if the need arises to make this configurable